src/lib/party.ts

```diff
@@ -482,10 +483,26 @@ type LazyControl = Control | LazySignature | LazyProof;
 
 class Party {
   body: Body;
-  authorization: LazyControl = {};
+  authorization: LazyControl;
+  account: Precondition.Account;
+  network: Precondition.Network;
 
-  constructor(body: Body) {
+  private isSelf: boolean;
+
+  constructor(body: Body, authorization?: LazyControl);
+  constructor(body: Body, authorization = {} as LazyControl, isSelf = false) {
     this.body = body;
+    this.authorization = authorization;
+    let { account, network } = Precondition.preconditions(this, isSelf);
+    this.account = account;
+    this.network = network;
+    this.isSelf = isSelf;
+  }
+
+  static clone(party: Party) {
```

Comment:
Before adding this, `Party`s were just cloned with `cloneCircuitValue` directly. I changed it because the new `account` and `network` fields aren't safely clone-able, due to how they use closures / are attached to a party instance. This new cloning will lose precondition-related metadata attached to a party (i.e. which account / network fields were accessed / constrained). this seems appropriate, because cloning a party also doesn't magically add it to the current transaction

Reply:
👍